### PR TITLE
Upgrade jest

### PR DESCRIPTION
**Pull request by ahutchings**
I was seeing a number of failures when running tests locally with node v6.2.0 (https://github.com/facebook/jest/issues/340). Upgrading to jest 12 fixed the original failures. Most of the other changes in this pull request are upgrading to the new test runner API - Jest 12 is based on Jasmine 2, which removed "runs()" and "waitsFor".

There is still one failing test:

``` javascript
The Bottom Scroll Preserving Behavior of the Bottom Upwards Display › it keeps the scroll attached to the bottom even when the element is capable of scrolling upwards
  - Expected undefined to equal 1200.
        at jasmine.buildExpectationResult (node_modules/jest-jasmine2/src/index.js:80:44)
        at Object.eval (__tests__/bottom_upwards_infinite_test.js:143:35)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:35:32)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:40:11)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:40:11)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:40:11)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:40:11)
        at Object.<anonymous> (node_modules/jest-jasmine2/src/jasmine-pit.js:40:11)
```

Later in the test output I see "Warning: `NaN` is an invalid value for the `height` css style property. Check the render method of `Infinite`." This seems to be related to the failing test. I tracked this back to an undefined value for the scrollable element's "scrollTop" property. I am not sure how to fix this issue, so any advice here would be appreciated.

Edit: Tests appear to be passing in Travis with node 5, so this might just be a node 6 issue.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ sudo: false
 language: node_js
 node_js:
   - 0.10
+  - "5"
```
Comment (DevSide) [inline .travis.yml]: node 4, 5, 6 and 7 ?

File: package.json [hunk 2]
```diff
@@ -70,9 +70,11 @@
     "gulp-sourcemaps": "^1.2.4",
     "gulp-uglify": "^1.0.1",
     "gulp-webserver": "^0.9.1",
-    "jest-cli": "0.4.19",
+    "jest-cli": "^12.1.1",
```
Comment (DevSide) [inline package.json]: "^20.0.1"

File: src/react-infinite.jsx [hunk 3]
```diff
@@ -245,8 +245,7 @@ var Infinite = React.createClass({
     newState.numberOfChildren = React.Children.count(computedProps.children);
     newState.infiniteComputer = infiniteHelpers.createInfiniteComputer(
       computedProps.elementHeight,
-      computedProps.children,
-      computedProps.displayBottomUpwards
```
Comment (DevSide) [inline src/react-infinite.jsx]: :+1:

**Comment by garetht** [comment]
Thanks! Just upgraded to Jest 20, so this will no longer be necessary.

**Pull request closed without merging**